lib/internal/assert/assertion_error.js

```diff
@@ -47,6 +49,15 @@ function copyError(source) {
     __proto__: null,
     value: source.message,
   });
+  if (source.cause !== undefined) {
```

Comment:
We should support the use-case for `new Error('message', { cause: undefined })`
```suggestion
  if (ObjectPrototypeHasOwnProperty(source, 'cause')) {
```

Reply:
Will do.

Reply:
I've opened https://github.com/nodejs/node/pull/55738 for that

Reply:
Cool! TY. 🫶